### PR TITLE
Refs #3809 - Turning on some Lint cops

### PR DESCRIPTION
**Pull request by daviddavis**

### CODE REVIEW

File: app/controllers/concerns/foreman/controller/taxonomies_controller.rb [hunk 1]
```diff
@@ -201,8 +201,6 @@ def count_nil_hosts
     @count_nil_hosts = hosts_scope_without_taxonomy.count
   end
 
-  private
```
Comment (daviddavis) [inline app/controllers/concerns/foreman/controller/taxonomies_controller.rb:204]: Duplicates private from line 161

File: app/helpers/filters_helper_overrides.rb [hunk 2]
```diff
@@ -13,11 +13,11 @@ def self.search_path(class_or_engine_name)
     @@overrides[deconstantize_name(class_or_engine_name)].call(class_or_engine_name)
   end
 
-  private
-
   def self.deconstantize_name(name)
     name.include?('::') ? name.deconstantize : name
   end
 
+  private
```
Comment (daviddavis) [inline app/helpers/filters_helper_overrides.rb:20]: I moved the class method `deconstantize_name` out of the private section due to IneffectiveAccessModifier

File: app/models/nic/bmc.rb [hunk 3]
```diff
@@ -38,8 +38,6 @@ def ensure_physical
       true # don't stop validation chain
     end
 
-    private
```
Comment (daviddavis) [inline app/models/nic/bmc.rb:41]: Duplicates private from line 34.

File: app/models/setting.rb [hunk 4]
```diff
@@ -198,8 +198,6 @@ def self.convert_array_to_regexp(array)
     Regexp.new(array.map {|string| regexp_expand_wildcard_string(string) }.join('|'))
   end
 
-  private
```
Comment (daviddavis) [inline app/models/setting.rb:201]: Moved private here to line 252 (after the class methods).

File: app/models/subnet.rb [hunk 5]
```diff
@@ -294,8 +294,6 @@ def new_network_type(args)
     end
   end
 
-  private
```
Comment (daviddavis) [inline app/models/subnet.rb:297]: Duplicates private from line 228

**Comment by daviddavis** [comment]
If any of these class methods need to actually be private, let me know and I can make them private.

File: app/models/user.rb [hunk 6]
```diff
@@ -440,6 +422,24 @@ def self.try_to_auto_create_user(login, password)
     end
   end
 
+  private
```
Comment (dLobatog) [inline app/models/user.rb:425]: Wow, github's diff was kind of dumb here? I guess you just moved the `self` methods  but it identified the rest of the code as moved

**Comment by dLobatog** [comment]
@daviddavis Thanks! :smile: 

Let's give it til the end of today in case someone voices any concerns about enabling these cops, I'll advertise it in the channel

**Comment by daviddavis** [comment]
@dLobatog sounds good.

**Comment by daviddavis** [comment]
[test]

**Comment by daviddavis** [comment]
[test]

**Comment by dLobatog** [comment]
@daviddavis I removed your last [test] comment - no need to retrigger it again, the error is transient and testing it continually puts more load on Jenkins for no reason. It's fine, I'll merge as it is in a few minutes. :smile:

**Comment by dLobatog** [comment]
Merged as 2ea02ed86f81d94d1092fa94bd197882128cdb97, thanks @daviddavis!

**Pull request closed without merging**